### PR TITLE
[src] New phase features and transforms

### PR DESCRIPTION
**Pull request by mpariente**
This PR adds
- Delta and double Delta computation of torch tensors
- A common phase correction that can be useful. 

### Questions:
Should we add `instantaneous_frequency` and `group_delay` functions? How about `nn.Module`? All the transforms are functional but having them as module can be useful.

### CODE REVIEW

File: asteroid/filterbanks/transforms.py [hunk 1]
```diff
@@ -371,17 +385,104 @@ def ebased_vad(mag_spec, th_db: int = 40):
...
+        >>> # Compute instantaneous frequency
+        >>> inst_freq = compute_delta(phase, dim=-1)
+        >>> # Or group delay
+        >>> group_delay = compute_delta(phase, dim=-2)
+    """
+    if dim != -1:
+        return compute_delta(feats.transpose(-1, dim), dim=-1).transpose(-1, dim)
+    # First frame has nothing. Then each frame is the diff with the previous one.
+    delta = feats.new_zeros(feats.shape)
+    delta[..., 1:] = feats[..., 1:] - feats[..., :-1]
```
Comment (jonashaag) [inline asteroid/filterbanks/transforms.py:410]: Wild speculation; could this be more efficient using `F.pad` instead?

File: asteroid/filterbanks/transforms.py [hunk 2]
```diff
@@ -342,6 +345,17 @@ def from_mag_and_phase(mag, phase, dim: int = -2):
     return torch.cat([mag * torch.cos(phase), mag * torch.sin(phase)], dim=dim)
 
 
+# Alias
+from_polar = from_mag_and_phase
```
Comment (jonashaag) [inline asteroid/filterbanks/transforms.py:349]: We should also add these aliases in `complex_nn` IMO or stick to any of the two names everywhere.

File: asteroid/filterbanks/transforms.py (hunk 1 above)
Comment (mpariente) [reply asteroid/filterbanks/transforms.py:410]: Maybe. Doesn't seem important to me at this point but you can have a go :wink:

File: asteroid/filterbanks/transforms.py [hunk 3]
```diff
@@ -342,6 +345,17 @@ def from_mag_and_phase(mag, phase, dim: int = -2):
...
+
+
+def to_polar(spec: torch.Tensor, dim: int = -2) -> Tuple[torch.Tensor, torch.Tensor]:
```
Comment (popcornell) [inline asteroid/filterbanks/transforms.py:352]: much needed IMO thanks Manu

File: asteroid/filterbanks/transforms.py [hunk 4]
```diff
@@ -371,17 +385,104 @@ def ebased_vad(mag_spec, th_db: int = 40):
     return log_mag > (max_log_mag - th_db)
```
Comment (popcornell) [inline asteroid/filterbanks/transforms.py:385]: maybe for VAD we should add some simple hangover and hangbefore (easy way is by using median filtering)

Comment (mpariente) [reply asteroid/filterbanks/transforms.py:385]: Would you like to do it? 
Maybe we could simply rely on torchaudio's VAD? I didn't check it but it's probably better than ours, and scriptable.

File: asteroid/filterbanks/transforms.py (hunk 2 above)
Comment (jonashaag) [reply asteroid/filterbanks/transforms.py:349]: Hm ok this is harder than I thought 😬 so want to have 4 things:

- “good” (meaningful) names
- consistent with torchaudio 
- consistent within Asteroid
- compatible with current Asteroid names

We’d probably have to put a non trivial amount of effort into getting all of these right...

I don’t know what to do really.

Comment (mpariente) [reply asteroid/filterbanks/transforms.py:349]: See #335

**Comment by mpariente** [comment]
I think I'm going to make the changes I proposed in #335 after this PR is merged.

**Pull request merged**